### ISSUE TITLE
Long bibliography entry looks bad

### ISSUE THREAD

**Issue by mglisse** [doxygen/doxygen#9344]
**Describe the bug**
When an entry in the bibliography becomes too long, in html/citelist.html, the end of the entry gets split into an ordered list
```html
<dt><a class="anchor" id="CITEREF_key"></a>[1]</dt>
<dd><p class="startdd">Author. Title. In <em>Proceedings of the 24th annual conference on publishing proceedings from very interesting conferences</em>, Conference '22, pages 123&ndash;132, Address,</p><ol type="1">
<li>Publisher.</li>
</ol>
<p class="enddd"></p>
</dd>
```

**Expected behavior**
Continue the same as with shorter entries.

**Screenshots**
![Screenshot from 2022-05-17 22-56-06](https://user-images.githubusercontent.com/10097863/168909238-e61b0058-02a7-44ea-a8f0-689aafea31e5.png)

**To Reproduce**
test.c
```c
/** hello \cite key. */
void f(int i){}
```
biblio.bib
```bib
@inproceedings{key,
 author = {Author},
 title = {Title},
 booktitle = {Proceedings of the 24th annual conference on publishing proceedings from very interesting conferences},
 series = {Conference '22},
 year = {2022},
 pages = {123--132},
 numpages = {10},
 publisher = {Publisher},
 address = {Address}, 
}
```
Run `doxygen -g`, set `CITE_BIB_FILES = biblio.bib` and run `doxygen`.

**Version**
1.9.4 (self-compiled) or 1.9.1 (from distribution) on debian testing x86_64.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9345

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.5.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
